include/libtorrent/kademlia/dht_storage.hpp

```diff
@@ -214,6 +264,24 @@ namespace dht
 		//
 		virtual void tick() = 0;
 
+		// Exports the stored items (immutable and mutable).
+		//
+		// For implementers:
+		// In case you provide a custom implementation, take in consideration
+		// that the data could be used, saved or loaded and that means that
+		// you should take care of the amount of data you want to return. It's
+		// advisable that you save no more than ``dht_settings::max_dht_items``
+		// for each item's type.
+		virtual dht_storage_items save_items() const = 0;
+
+		// Loads the items in the storage.
+		//
+		// For implementers:
+		// It's recommended that you take in consideration the ``last_seen``
+		// field in the items during the load phase. Similarly you should not
+		// store more than ``dht_settings::max_dht_items``.
+		virtual void load_items(dht_storage_items items) = 0;
```

Comment:
do you think there's value in being able to load items after construction? or load items more than once? In some sense it smells a little bit like two-phase initialization. Do you think it would be inconvenient to load in the constructor instead?